src/nbl/video/CVKLogicalDevice.h

```diff
@@ -178,52 +802,382 @@ class CVKLogicalDevice final : public ILogicalDevice
+        core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& layout3 = nullptr) override
     {
-        return nullptr;
+        constexpr uint32_t MAX_PC_RANGE_COUNT = 100u;
+        constexpr uint32_t MAX_DESCRIPTOR_SET_LAYOUT_COUNT = 4u; // temporary max, I believe
+
+        const core::smart_refctd_ptr<IGPUDescriptorSetLayout> tmp[] = { layout0, layout1, layout2,
+            layout3 };
+
+        VkDescriptorSetLayout vk_dsLayouts[MAX_DESCRIPTOR_SET_LAYOUT_COUNT];
+        uint32_t dsLayoutCount = 0u;
+        for (uint32_t i = 0u; i < MAX_DESCRIPTOR_SET_LAYOUT_COUNT; ++i)
+        {
+            if (tmp[i] && (tmp[i]->getAPIType() == EAT_VULKAN))
+                vk_dsLayouts[dsLayoutCount++] = static_cast<const CVulkanDescriptorSetLayout*>(tmp[i].get())->getInternalObject();
+        }
+
+        const auto pcRangeCount = std::distance(_pcRangesBegin, _pcRangesEnd);
+        assert(pcRangeCount <= MAX_PC_RANGE_COUNT);
+        VkPushConstantRange vk_pushConstantRanges[MAX_PC_RANGE_COUNT];
+        for (uint32_t i = 0u; i < pcRangeCount; ++i)
+        {
+            const auto pcRange = _pcRangesBegin + i;
+
+            vk_pushConstantRanges[i].stageFlags = static_cast<VkShaderStageFlags>(pcRange->stageFlags);
+            vk_pushConstantRanges[i].offset = pcRange->offset;
+            vk_pushConstantRanges[i].size = pcRange->size;
+        }
+
+        VkPipelineLayoutCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_PIPELINE_LAYOUT_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // pNext must be NULL
+        vk_createInfo.flags = static_cast<VkPipelineLayoutCreateFlags>(0); // flags must be 0
+        vk_createInfo.setLayoutCount = dsLayoutCount;
+        vk_createInfo.pSetLayouts = vk_dsLayouts;
+        vk_createInfo.pushConstantRangeCount = pcRangeCount;
+        vk_createInfo.pPushConstantRanges = vk_pushConstantRanges;
+                
+        VkPipelineLayout vk_pipelineLayout;
+        if (vkCreatePipelineLayout(m_vkdev, &vk_createInfo, nullptr, &vk_pipelineLayout) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanPipelineLayout>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), _pcRangesBegin, _pcRangesEnd,
+                std::move(layout0), std::move(layout1), std::move(layout2), std::move(layout3),
+                vk_pipelineLayout);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
-    core::smart_refctd_ptr<IGPUComputePipeline> createGPUComputePipeline_impl(IGPUPipelineCache* _pipelineCache,
-        core::smart_refctd_ptr<IGPUPipelineLayout>&& _layout, core::smart_refctd_ptr<IGPUSpecializedShader>&& _shader) override
+    // For consistency's sake why not pass IGPUComputePipeline::SCreationParams as
+    // only second argument, like in createGPUComputePipelines_impl below? Especially
+    // now, since I've added more members to IGPUComputePipeline::SCreationParams
```

Comment:
go for it, if @Crisspl approves

Reply:
@achalpandeyy refresh my memory

Reply:
It is about making the the following two function signatures similar:

Separate args to the function `createGPUComputePipeline_impl` can be wrapped up into just one `IGPUComputePipeline::SCreationParams`, like for `createGPUComputePipelines_impl` which has the signature

`bool createGPUComputePipelines_impl(IGPUPipelineCache* pipelineCache,
        core::SRange<const IGPUComputePipeline::SCreationParams> createInfos,
        core::smart_refctd_ptr<IGPUComputePipeline>* output) override`